### DISCUSSION TITLE
How to specify job params available to all ops?

### DISCUSSION THREAD

**Discussion by clayheaton** [dagster-io/dagster#19564] [Q&A] [answered]
This might be a question related to migrating to the more pythonic config system. I've looked at other answers but continue to struggle to get this to work.

I need to specify a job that has params that can be set in the Dagster UI. Instead of these params being available only to a single specified op or asset, I would like them available to all ops or assets in the job. Here's how I've done the asset-specific ones in the past:

```python
my_job = define_asset_job(
    name="My_Job",
    description="About my job",
    selection=AssetSelection.groups(MY_ASSET_GROUP_NAME),
    config={
        "ops": {
            "MY_ASSET": {
                "config": {
                    "param1": "value1",
                    "param2": "value2",
                }
            }
        }
    },
)
```

When launching the job in the UI, the Launchpad appears, `param1` and `param2` can be edited, but then they are only available to `MY_ASSET`.

In `MY_ASSET`, I can easily get `param1` and `param2` with

```python
@asset(
    config_schema={"param1": str, "param2": str},
)
def MY_ASSET(context: OpExecutionContext) -> None:
    context.log.info("param1: " + context.op_config["param1"])
    context.log.info("param2: " + context.op_config["param2"])
```

What I'd really like is to do something like this:

```python
my_job = define_asset_job(
    name="My_Job",
    description="About my job",
    selection=AssetSelection.groups(MY_ASSET_GROUP_NAME),
    config={
        "job":{
            "param1": "value1",
            "param2": "value2",
        },
    },
)
```

... and then be able to access it in any of the `MY_ASSET_GROUP_NAME` assets with `context.job_config["param1"]` or a similarly simple approach. The use case here is that each time the job is executed, the user must input `user_input_value1` and `user_input_value2` in the Dagster UI and those values will be used to parameterized some queries, etc. 

Since this seems like a simple and common use case -- can anybody provide me with the proper syntax for specifying this as the `config` for the job? I've tried a number of variations (based on the documentation) and they all crash the code location b/c of an invalid config.

Incidentally, the documentation for `define_asset_job` states:

> If a dictionary is provided, then it must conform to the standard config schema, and it will be used as the job’s run config for the job whenever the job is executed. The values provided will be viewable and editable in the Dagster UI, so be careful with secrets.

> If a [ConfigMapping](https://docs.dagster.io/_apidocs/config#dagster.ConfigMapping) object is provided, then the schema for the job’s run config is determined by the config mapping, and the ConfigMapping, which should return configuration in the standard format to configure the job.

Ok - but I'm having trouble finding documentation on `the standard config schema` or `standard format` mentioned.

**Comment by alangenfeld** [comment]
The best approach in this context is probably to make a "global config" *resource* and add that resource too any assets that need to access it. You can see more about this approach in this discussion https://github.com/dagster-io/dagster/discussions/3213

**Reply by clayheaton** [reply]
Will that global config resource appear in Launchpad so that people starting the job are able to change the values?

**Reply by alangenfeld** [reply, marked as answer]
yep, you can use a `ConfigurableResource` and `configure_at_launch()` 
https://docs.dagster.io/concepts/resources#configuring-resources-at-launch-time
